### PR TITLE
Pull release notes from remote (cdn.dl.k8s.io)

### PR DESCRIPTION
**Pull request by ashnehete**
This PR adds functionality to fetch release notes stored on Google Storage bucket using cdn.dl.k8s.io. 
Reference: [kubernetes/release#1087](https://github.com/kubernetes/release/issues/1087)

It first fetches the list of release notes from [cdn.dl.k8s.io/release/release-notes-index.json](https://cdn.dl.k8s.io/release/release-notes-index.json). Then subsequently extracts all the release notes.

### CODE REVIEW

**Comment by saschagrunert** [comment]
Thank you for this PR @ashnehete, unfortunately is the CI not happy.

**Comment by ashnehete** [comment]
Fixed the problems with tests and have expanded the e2e tests to encompass the new features. Though I was wondering if I should make the e2e tests more comprehensive?
Would love any reviews on the code.

**Review by saschagrunert — APPROVED** [review]

**Comment by saschagrunert** [comment]
> Though I was wondering if I should make the e2e tests more comprehensive? 

Yeah that would be a great follow-up! :)

File: src/environments/index-url.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export const indexUrl = 'https://cdn.dl.k8s.io/release/release-notes-index.json';
```
[the remark this answers is not in the document]
Comment (ashnehete) [reply src/environments/index-url.ts:1]: My first thought was the same, but upon inspection I realised 2 things:
1. `dl.k8s.io` fails CORS request but `cdn.dl.k8s.io` has CORS enabled.
2. `dl.k8s.io` does a 302 redirect to `cdn.dl.k8s.io`.

I'm currently trying to figure out how to enable CORS on `dl.k8s.io`. Please point me in the direction if you have any leads.

[the remark this answers is not in the document]
Comment (xmudrii) [reply src/environments/index-url.ts:1]: @dims @ameukam @BenTheElder Can you please provide some feedback on this?

[the remark this answers is not in the document]
Comment (ashnehete) [reply src/environments/index-url.ts:1]: I think the relevant section for setting the redirection is: [k8s.io/apps/k8s-io/configmap-nginx.yaml#L181-L194](https://github.com/kubernetes/k8s.io/blob/4930336672f4de53982e6e41baeb26c29fbe577f/apps/k8s-io/configmap-nginx.yaml#L181-L194).

I see that the `cdn.dl.k8s.io` at [services.tf#L83-L90](https://github.com/kubernetes/k8s.io/blob/main/infra/fastly/terraform/dl.k8s.io/services.tf#L83-L90) enables CORS for that domain.

[the remark this answers is not in the document]
Comment (dims) [reply src/environments/index-url.ts:1]: that smells like the right spot(s) @ashnehete

[the remark this answers is not in the document]
Comment (BenTheElder) [reply src/environments/index-url.ts:1]: cdn.dl.k8s.io is an alias to a specific bucket. dl.k8s.io is a more complex redirect to N backends, including CI artifacts that are NOT in the more secure release bucket.

**Pull request merged**